src/mlpack/methods/ann/layer/batchnorm_impl.hpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * @file batchnorm_impl.hpp
+ * @author Praveen Ch
+ *
+ * Implementation of the Batch Normalization Layer.
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LAYER_BATCHNORM_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_BATCHNORM_IMPL_HPP
+
+// In case it is not included.
+#include "batchnorm.hpp"
+
+namespace mlpack {
+namespace ann { /** Artificial Neural Network. */
+
+template <typename InputDataType, typename OutputDataType>
+BatchNorm<InputDataType, OutputDataType>::BatchNorm(
+    const size_t size, const double eps = 0.001) : size(size), eps(eps)
```

Comment:
If someone includes just the hpp file, all it sees is the header file, since there is no default value for the second parameter, the code should fail if you pass just the first parameter. I wonder why g++ is fine with it.